### ISSUE TITLE
TrailingComma causes false positive for UnusedPrivateMember

### ISSUE THREAD

**Issue by cuongvo** [detekt/detekt#5486]
I understand `TrailingComma` is deprecated and moved into two separate top level rules with the latest 1.22 RC releases. I have not verified this behavior exists there.

## Expected Behavior
Enable `TrailingComma` (v1.21.0) should not cause `UnusedPrivateMember` to treat members requiring a trailing comma as unused and trigger a false positive.

## Observed Behavior
Enable `TrailingComma` support in formatting rules and enable auto correcting. Notice trailing commas violations are reported but false positive `UnusedPrivateMember` violations are also reported. 

When ` excludeCorrectable: true` is set, this will cause a build error due to the `UnusedPrivateMember` false positives.

This only occurs on the initial run where the files are auto corrected to include trailing commas. Once the trailing commas exist, subsequent runs will not trigger `UnusedPrivateMember` false positives.

## Steps to Reproduce
```
  TrailingComma:
    active: true
    autoCorrect: true
    allowTrailingComma: true
    allowTrailingCommaOnCallSite: true
```
Run Detekt. Notice `UnusedPrivateMember` violations for false positives.

Disable `TrailingComma`

Run Detekt. `UnusedPrivateMember` violations no longer exist.

## Your Environment

* Version of detekt used: 1.21.0

**Comment by cortinico** [comment]
> I understand `TrailingComma` is deprecated and moved into two separate top level rules with the latest 1.22 RC releases. I have not verified this behavior exists there.

Could you try to do so? The latest RC is using the latest version of KtLint which might have fixed this

**Comment by cuongvo** [comment]
I can check and update this issue once I confirm. I also updated the original issue to include more details:

> This only occurs on the initial run where the files are auto corrected to include trailing commas. Once the trailing commas exist, subsequent runs will not trigger UnusedPrivateMember false positives.

Unsure if that helps in figuring out if this would occur in the RCs until I validate.
